cmd/provider/add.go

```diff
@@ -45,30 +54,48 @@ func NewAddCmd(flags *flags.GlobalFlags) *cobra.Command {
 
 	addCmd.Flags().BoolVar(&cmd.SingleMachine, "single-machine", false, "If enabled will use a single machine for all workspaces")
 	addCmd.Flags().StringVar(&cmd.Name, "name", "", "The name to use for this provider. If empty will use the name within the loaded config")
+	addCmd.Flags().StringVar(&cmd.FromExisting, "from-existing", "", "The name of an existing provider to use as a template. Needs to be used in conjunction with the --name flag")
```

Comment:
Here we say that we need to use it with a `--name` flag, but we should also add this check in the next section, if `--name` is not specified and `--from-existing` is specified, we should error out and warn the user about this

Reply:
that should be handled in the `PreRunE` function in line 37 👍